### PR TITLE
Added Google Calendar Features

### PR DESCRIPTION
**Pull request by HarshwardhanSingh**
Let me know if there are any issues.

### CODE REVIEW

File: app/controllers/calendars_controller.rb [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    client = User.connectCal(current_user.oauth_token, current_user.refresh_token)
+    service = client.discovered_api('calendar' , 'v3')
+    @response = client.execute(api_method: service.events.list,
+            parameters: { 'calendarId' => Base64.urlsafe_decode64(params[:id]),
+              'timeMin': Time.now.beginning_of_month.iso8601,
+              'showDeleted': false,
+              'singleEvents': true,
+              'maxResults': 10,
+              'orderBy': 'startTime'},
+                headers: { 'Content-Type' => 'application/json' })
```
Comment (revathskumar) [inline app/controllers/calendars_controller.rb]: Indentation.

File: app/controllers/calendars_controller.rb [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    render 'show'
+  end
+
+  def new
+  end
+
+  # def createCalendar
+  def create
+    @calendar = {'summary' => params[:summary]}
+    client = User.connectCal(current_user.oauth_token, current_user.refresh_token)
```
Comment (revathskumar) [inline app/controllers/calendars_controller.rb]: You can move `User.connectCal(current_user.oauth_token, current_user.refresh_token)` to a `:before_action` callback.

File: app/controllers/calendars_controller.rb (hunk 1 above)
Comment (iffyuva) [inline app/controllers/calendars_controller.rb]: time to integrate rubocop!

/cc @goromlagche

File: config/routes.rb [hunk 3]
```diff
@@ -12,11 +12,22 @@
...
 
+  resources :calendars do
+    resources :events
+  end
+
+  # Manage Calendar settings
+  get '/calendars/:id/settings', to: 'calendars#settings', as: 'calendar_settings'
+
+  # Invite users to calendar
+  post '/calendars/:id/invite', to: 'calendars#invite', as: 'invite_user'
```
Comment (revathskumar) [inline config/routes.rb]: calendar `settings` and  `invite` can go into the  `resources :calendars` block.

File: app/controllers/calendars_controller.rb (hunk 2 above)
Comment (HarshwardhanSingh) [inline app/controllers/calendars_controller.rb]: or we could create a new method in the calendar controller and put the method to before_action instead.

File: app/controllers/calendars_controller.rb (hunk 1 above)
Comment (HarshwardhanSingh) [inline app/controllers/calendars_controller.rb]: Should I fix the indentation issues everywhere else also?

**Comment by revathskumar** [comment]
Merging this as of now. Need Lot of fixes. will do in coming tasks.

**Pull request merged**